src/loaders/FileLoader.js

```diff
@@ -55,225 +53,178 @@ class FileLoader extends Loader {
+			// TODO-DefinitelyMaybe: An abort controller could be added within a future PR
+		} );
 
-						for ( let i = 0; i < data.length; i ++ ) {
-
-							view[ i ] = data.charCodeAt( i );
-
-						}
-
-						if ( responseType === 'blob' ) {
-
-							response = new Blob( [ view.buffer ], { type: mimeType } );
-
-						} else {
-
-							response = view.buffer;
-
-						}
+		// start the fetch
+		fetch( req )
+			.then( response => {
 
-						break;
+				if ( response.status === 200 || response.status === 0 ) {
 
-					case 'document':
-
-						const parser = new DOMParser();
-						response = parser.parseFromString( data, mimeType );
-
-						break;
-
-					case 'json':
-
-						response = JSON.parse( data );
-
-						break;
-
-					default: // 'text' or other
-
-						response = data;
+					// Some browsers return HTTP Status 0 when using non-http protocol
+					// e.g. 'file://' or 'data://'. Handle as success.
 
-						break;
+					if ( response.status === 0 ) {
 
-				}
+						console.warn( 'THREE.FileLoader: HTTP Status 0 received.' );
 
-				// Wait for next browser tick like standard XMLHttpRequest event dispatching does
-				setTimeout( function () {
+					}
 
-					if ( onLoad ) onLoad( response );
+					const callbacks = loading[ url ];
+					const reader = response.body.getReader();
+					const contentLength = response.headers.get( 'Content-Length' );
+					const total = contentLength ? parseInt( contentLength ) : 0;
+					const lengthComputable = total != 0;
+					let loaded = 0;
 
-					scope.manager.itemEnd( url );
+					// periodically read data into the new stream tracking while download progress
+					return new ReadableStream( {
+						start( controller ) {
 
-				}, 0 );
+							readData();
 
-			} catch ( error ) {
+							function readData() {
 
-				// Wait for next browser tick like standard XMLHttpRequest event dispatching does
-				setTimeout( function () {
+								reader.read()
+									.then( ( { done, value } ) => {
 
-					if ( onError ) onError( error );
+										if ( done ) {
 
-					scope.manager.itemError( url );
-					scope.manager.itemEnd( url );
+											controller.close();
 
-				}, 0 );
+										} else {
 
-			}
+											loaded += value.byteLength;
 
-		} else {
+											const event = new ProgressEvent( 'progress', { lengthComputable, loaded, total } );
+											for ( let i = 0, il = callbacks.length; i < il; i ++ ) {
 
-			// Initialise array for duplicate requests
+												const callback = callbacks[ i ];
+												if ( callback.onProgress ) callback.onProgress( event );
 
-			loading[ url ] = [];
+											}
 
-			loading[ url ].push( {
-
-				onLoad: onLoad,
-				onProgress: onProgress,
-				onError: onError
+											controller.enqueue( value );
+											readData();
 
-			} );
+										}
 
-			request = new XMLHttpRequest();
+									} );
 
-			request.open( 'GET', url, true );
+							}
 
-			request.addEventListener( 'load', function ( event ) {
+						}
+        	} );
```

Comment:
Looks like the indentation here isn't correct

Reply:
unsure? I simply run a `lint-fix` before submitting. Does look off tho.

Reply:
There are spaces in the indentation. I'm surprised the linter doesn't catch it in the CI check, though.